iothub/device/src/Transport/Amqp/AmqpConnectionPool.cs

```diff
@@ -36,6 +36,21 @@ internal class AmqpConnectionPool : IAmqpUnitManager
                 {
                     AmqpConnectionHolder[] amqpConnectionHolders = ResolveConnectionGroup(deviceIdentity);
                     amqpConnectionHolder = ResolveConnectionByHashing(amqpConnectionHolders, deviceIdentity);
+
+                    // For group sas token authenticated devices over a multiplexed connection, the TokenRefresher
+                    // of the first client connecting will be used for generating the group sas tokens
+                    // and will be associated with the connection itself.
+                    // For this reason, if the device identity of the client is not the one associated with the
+                    // connection, the associated TokenRefresher can be safely disposed.
+                    // Note - This does not cause any identity related issues since the group sas tokens are generated
+                    // against "{IoT hub name}.azure-devices.net/" as the intended audience (without the "device Id").
```

Comment:
Not if US Gov't cloud, or in the future with vanity domains.

Reply:
In that case, I'll edit to to simply say "hub hostname".